### PR TITLE
Tvlatas/vz 2398 3

### PR DESCRIPTION
**Pull request by agv426**
# Description

This is related to the analysis tooling user facing work. This adds:

- More image pull detections
- LB detection cases
- More testing (I didn't enable coverage as it brings the rest of Verrazzano down slightly from 75.7% to 75.2%). Will enable next sprint)
- Text placeholders updated to more readable messages (will evolve, need to go over with writers)
- stake in the ground for advice documentation (these will evolve, we may want to eventually have them in the docs instead of github)
- Other misc tweaks

Addresses VZ-2398

# Checklist 

As the author of this PR, I have:

- [X] Checked that I included or updated copyright and license notices in all files that I altered
- [X] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: tools/analysis/advice/IngressOciIPLimitExceeded.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Summary
+Analysis detected that the Verrazzano install failed while installing the NGINX Ingress Controller.
+
+The root cause appears to be an OCI IP limit has been reached
```
Comment (michael-cico) [inline tools/analysis/advice/IngressOciIPLimitExceeded.md:4]: probably say "IP Address limit" to be more explicit.  But probably can clean up on the next pass.

File: tools/analysis/internal/util/cluster/crds.go [hunk 2]
```diff
@@ -4,12 +4,7 @@
...
-import (
-	"github.com/verrazzano/verrazzano/tools/analysis/internal/util/json"
-	"go.uber.org/zap"
-)
-
 // GetCustomResource gets a customResource list
-func GetCustomResources(log *zap.SugaredLogger, path string) (customResources interface{}, err error) {
-	return json.GetJSONDataFromFile(log, path)
-}
+//func GetCustomResources(log *zap.SugaredLogger, path string) (customResources interface{}, err error) {
```
Comment (michael-cico) [inline tools/analysis/internal/util/cluster/crds.go:8]: probably clean this up on the next pass.

**Review by michael-cico — DISMISSED** [review]

File: tools/analysis/internal/util/cluster/crds.go (hunk 2 above)
Comment (agv426) [reply tools/analysis/internal/util/cluster/crds.go:8]: Yeah, this is a placeholder for coming work actually (same with CRD)

File: tools/analysis/advice/IngressOciIPLimitExceeded.md (hunk 1 above)
Comment (agv426) [reply tools/analysis/advice/IngressOciIPLimitExceeded.md:4]: Yeah, This is the initial stake in the ground, I'm going to be doing a pass on the content and working with the doc folks on this in the next sprint.

**Review by michael-cico — APPROVED** [review]

**Pull request merged**